pkg/strategies/loop_pointer_arithmetic.go

```diff
@@ -0,0 +1,287 @@
+package strategies
+
+import (
+	. "buzzer/pkg/ebpf/ebpf"
+	"buzzer/pkg/rand"
+	"buzzer/pkg/units/units"
+	btfpb "buzzer/proto/btf_go_proto"
+	epb "buzzer/proto/ebpf_go_proto"
+	fpb "buzzer/proto/ffi_go_proto"
+	pb "buzzer/proto/program_go_proto"
+	"fmt"
+)
+
+func NewLoopPointerArithmeticStrategy() *LoopPointerArithmetic {
+	return &LoopPointerArithmetic{isFinished: false}
+}
+
+// LoopPointerArithmetic is a strategy meant for testing, users can generate Arbitrary
+// programs and then the results of the verifier will be displayed on screen.
+type LoopPointerArithmetic struct {
+	isFinished        bool
+	mapFd             int
+	programCount      int
+	validProgramCount int
+	log               string
+}
+
+func types() []*btfpb.BtfType {
+	types := []*btfpb.BtfType{}
+
+	// 1: Func_Proto
+	types = append(types, &btfpb.BtfType{
+		NameOff:    0x0,
+		Info:       SetTypeInfo(0, btfpb.BtfKind_FUNCPROTO, false),
```

Comment:
I think for type info, it would be slightly more readable if in the proto we declare it as another message 

```
message TypeInfo {
  int32 vlen = 1 
  ...
}
```

And then we fill in the details here as needed instead of calling in this function. That way the people building the proto message know exactly what they are passing.

Then we can make your SetTypeInfo function (renamed to EncodeTypeInfo as I suggested above) receive a TypeInfo message and do the encoding to an int at the encode btf time.

Reply:
Done

Reply:
Done